### DISCUSSION TITLE
Support dynamic routes in static export, with traditional SPA-like routing

### DISCUSSION THREAD

**Discussion by jroitgrund** [vercel/next.js#35503] [Ideas]
### Describe the feature you'd like to request

next.js should support a static export where `getStaticPaths` is not defined but dynamic routes still work, just like in a traditional SPA: the server always serves the same index.html entrypoint, but routing is done on the client side.

### Describe the solution you'd like

Users would configure their webserver to serve `index.html` file for every path, and routing would happen just like it typically does on SPAs. on the front-end. For example:
* app contains `pages/user/[id].tsx` (user page)
* build static export, which generates `_next/static/chunks/pages/user/[id].js` and a single entry-point, `index.html`
* browser requests `GET /user/foo`, webserver hosting the static export responds with the contents of `index.html`
* the client-side code in parses the `/user/foo` route, loads `_next/static/chunks/pages/user/[id].js` and renders the user page.

Note that this already *almost* works. If you configure your webserver to serve the right files (e.g. `/users/[id.html]` for requests to `/users/<id>`) then the app works entirely as expected, including all client-side routing.

### Describe alternatives you've considered

Not using next.js, but this would be a shame! Even though none of my content is static, next.js still offers a great development environment with fast TS and CSS-in-JS compilation and bundling, code splitting, and routing.

**Comment by Ryomasao** [comment]
There may be some problems, but it seems to work.
※ we need to aggregate all requests to index.html


`ResolvePath.tsx`
```tsx
export const ResolvePath = (props: Props) => {
  // enabled only production
  const [resolved, setResolved] = useState(
    process.env.NODE_ENV === 'development',
  )
  const router = useRouter()
  const currentPath = router.asPath
  useEffect(() => {
    if (!resolved) {
      router.push(currentPath)
      setResolved(true)
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
  if (resolved) {
    return <>{props.children}</>
  }
  return null
}
```
`_app.tsx`
```tsx
<ResolvePath>
 {props.children}
</ResolvePath>
```

**Comment by rs-8** [comment]
Can we use nextjs for building spa apps and deploy it with nginx?
Currently we need to make custom workaround and I'm not sure that it will be work without node runtime at server
